projects/go-lib/src/lib/components/go-toast/go-toast.component.spec.ts

```diff
@@ -18,10 +18,77 @@ describe('GoToastComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(GoToastComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('ngOnInit', () => {
+    it('sets a default statusClass if no type is passed in', () => {
+      fixture.detectChanges();
+
+      expect(component.statusClass).toEqual('go-toast-status--neutral');
+    });
+
+    it('sets a statusClass if type is passed in', () => {
+      component.type = 'positive';
+      fixture.detectChanges();
+
+      expect(component.statusClass).toEqual('go-toast-status--positive');
+    });
+
+    it('sets a default icon if no type is passed in', () => {
+      fixture.detectChanges();
+
+      expect(component.icon).toEqual('notifications_none');
+    });
+
+    it('should set an icon if type is passed in', () => {
+      component.type = 'positive';
+      fixture.detectChanges();
+
+      expect(component.icon).toEqual('done');
+    });
+  });
+
+  describe('dismiss', () => {
+    it('emits the handleClick event', () => {
+      spyOn(component.handleDismiss, 'emit');
+      spyOn(component, 'dismiss').and.callThrough();
+
+      component.dismissable = true;
+      fixture.detectChanges();
+
+      const goToastTemplate: HTMLElement = fixture.nativeElement;
+      const buttonElement: HTMLElement = goToastTemplate.querySelector('.go-toast-dismiss__icon');
+
+      buttonElement.dispatchEvent(new Event('click'));
+      fixture.detectChanges();
+
+      expect(component.dismiss).toHaveBeenCalled();
+      expect(component.handleDismiss.emit).toHaveBeenCalled();
+    });
+  });
+
+  describe('the template', () => {
+    it('renders dismiss button if dismissable is true', () => {
```

Comment:
Does this test pass?
`dismissable` is false be default, so to test this case correctly I think the assertion needs to be:
`expect(goToastDismissButton).not.toBeNull();`